pkgs/development/python-modules/torch/default.nix

```diff
@@ -290,6 +290,22 @@ in buildPythonPackage rec {
   # error: cast from ... to ... converts to incompatible function type [-Werror,-Wcast-function-type-strict]
   ++ lib.optionals (stdenv.cc.isClang && lib.versionAtLeast stdenv.cc.version "16") [
     "-Wno-error=cast-function-type-strict"
+  # Suppresses the most spammy warnings.
+  # This is mainly to fix https://github.com/NixOS/nixpkgs/issues/266895.
+  ] ++ lib.optionals rocmSupport [
+    "-Wno-#warnings"
+    "-Wno-cpp"
+    "-Wno-unknown-warning-option"
+    "-Wno-ignored-attributes"
+    "-Wno-unused-result"
+    "-Wno-deprecated-declarations"
+    "-Wno-defaulted-function-deleted"
+    "-Wno-pass-failed"
+  ] ++ [
+    "-Wno-unused-command-line-argument"
+    "-Wno-maybe-uninitialized"
+    "-Wno-uninitialized"
+    "-Wno-array-bounds"
```

Comment:
```suggestion
    "-Wno-stringop-overflow"
```

Reply:
Is there a `-Wall` somewhere in the code?

Reply:
This in postPatch
`find . -type f -iname "*.cmake" -o -iname "CMakeLists.txt" -o -iname "Makefile" -exec sed -i '/-Wall/d' {} \;`

will get rid of the `Wall`'s but many of these projects enable individual `-W` too so it's not very useful

Reply:
I only got rid of the absolute worst spammy warnings, I see no reason to get rid of them all.

Reply:
Also I think getting rid of all the warning options could be prone to breakage, so best avoided.